pkg/cmd/experimental/bundlesecret/bundle-secret.go

```diff
@@ -0,0 +1,117 @@
+package bundlesecret
+
+//export OPENSHIFTCONFIG=./openshift.local.certificates/admin/.kubeconfig
+import (
+	"encoding/base64"
+	"encoding/binary"
+	"fmt"
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kcmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/spf13/cobra"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+// MaxFileSize?
+const MaxSecretSize = 1 * 1024 * 1024
+const MaxFileSize = 1 * 1024 * 1024
+
+func NewCmdBundleSecret(f *clientcmd.Factory, parentName, name string, out io.Writer) *cobra.Command {
+	var sourcelist util.StringList
+	cmd := &cobra.Command{
+		Use:   "bundle-secret NAME -f SOURCENAME",
+		Short: "bundle data from a source list to load kubernetes secret",
+		Long:  "bundle data from a source list to load kubernetes secret",
+		Run: func(cmd *cobra.Command, args []string) {
+			var err error
+			if len(args) == 0 {
+				err = fmt.Errorf("Must specify NAME as an argument")
+				checkErr(err)
+			}
+			objname := args[0]
+			if len(sourcelist) == 0 {
+				err = fmt.Errorf("Must specify files and/or directories to gather data from.")
+				checkErr(err)
+			}
+			for _, isspace := range sourcelist {
+				if isspace == "" {
+					err = fmt.Errorf("No spaces allowed.  Separate source names with commas, no spaces.")
+					checkErr(err)
+				}
+			}
+			secretdata := make(map[string][]byte)
+			for _, sourceitem := range sourcelist {
+				sourceitem = strings.TrimSuffix(sourceitem, "/")
+				info, err := os.Stat(sourceitem)
+				checkErr(err)
+				if info.IsDir() {
+					filelist, err := ioutil.ReadDir(sourceitem)
+					checkErr(err)
+					for _, item := range filelist {
+						if !item.IsDir() {
+							readValidFiles(sourceitem, item.Name(), item.Size(), secretdata)
+						}
+					}
+				} else {
+					readValidFiles(sourceitem, info.Name(), info.Size(), secretdata)
+				}
+			}
+			secretobj := kapi.Secret{
+				ObjectMeta: kapi.ObjectMeta{Name: objname},
+				Data:       secretdata,
+			}
+			totalsize := 0
+			for _, bytes := range secretdata {
+				totalsize += len(bytes)
+			}
+			if totalsize == 0 {
```

Comment:
this should be `len(secretobj.Data) == 0`... we shouldn't care if they want to make empty files part of a secret